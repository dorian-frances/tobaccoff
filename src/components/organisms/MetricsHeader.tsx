import { StyleSheet } from 'react-native';
import InfoText from '../atoms/texts/InfoText';

type MetricsHeaderProps = {
  text: string;
};

const MetricsHeader = ({ ...props }: MetricsHeaderProps) => {
  return <InfoText text={props.text} />;
};

const styles = StyleSheet.create({});

export default MetricsHeader;
