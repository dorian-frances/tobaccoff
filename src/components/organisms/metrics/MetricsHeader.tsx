import React, { useState } from 'react';
import TextInfo, { TextInfoProps } from '../../atoms/texts/TextInfo';
import IconButton from '../../atoms/buttons/ButtonIcon';
import { StyleSheet, View } from 'react-native';
import { Menu } from 'react-native-paper';
import { FontsEnum } from '../../../assets/fonts/fonts.enum';
import { fontPixel, fontStyles } from '../../../utils/font-scale.utils';
import { ColorsEnum } from '../../../assets/colors/colors.enum';

type MetricsHeaderProps = {
  infoTextProps: TextInfoProps;
  onPress: () => void;
};

const MetricsHeader = ({ infoTextProps, onPress }: MetricsHeaderProps) => {
  const [visible, setVisible] = useState(false);

  const closeMenu = () => setVisible(false);
  const openMenu = () => setVisible(true);
  return (
    <View style={styles.container}>
      <View style={styles.dateInfo}>
        <TextInfo {...infoTextProps} />
      </View>
      <View style={styles.buttonsContainer}>
        <Menu
          visible={visible}
          onDismiss={closeMenu}
          anchor={<IconButton icon={'dots-vertical'} onPress={openMenu} />}
          contentStyle={{
            backgroundColor: ColorsEnum.PRIMARY_95,
          }}
        >
          <Menu.Item
            title={'RÃ©initialiser mon compteur'}
            onPress={() => {
              onPress();
              closeMenu();
            }}
            leadingIcon={'restart'}
            titleStyle={{
              fontSize: fontStyles.small,
              fontFamily: FontsEnum.MEDIUM,
            }}
          />
        </Menu>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  dateInfo: {
    justifyContent: 'center',
  },
  settingsButton: {},
  buttonsContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
});
export default MetricsHeader;
