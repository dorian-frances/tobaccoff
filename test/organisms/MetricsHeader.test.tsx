import { render } from '@testing-library/react-native';
import MetricsHeader from '../../src/components/organisms/MetricsHeader';
import React from 'react';

describe('<MetricsHeader/>', () => {
  const fakeText = 'fakeText';
  it('should render correctly', () => {
    const component = render(
      <MetricsHeader infoTextProps={{ text: fakeText }} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
