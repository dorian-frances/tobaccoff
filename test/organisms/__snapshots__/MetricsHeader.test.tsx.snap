// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetricsHeader/> should render correctly 1`] = `
<Text
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "fontFamily": "System",
        "fontWeight": "400",
        "letterSpacing": 0,
      },
      {
        "writingDirection": "ltr",
      },
      {
        "fontFamily": "Montserrat-Medium",
        "fontSize": 10,
      },
    ]
  }
>
  fakeText
</Text>
`;
